src/cascadia/Remoting/Peasant.cpp

```diff
@@ -289,4 +290,24 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                           TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
                           TraceLoggingKeyword(TIL_KEYWORD_TRACE));
     }
+
+    // Method Description:
+    // - Request and return the window layout from the current TerminalPage
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - the window layout as a json string
+    hstring Peasant::GetWindowLayout()
+    {
+        auto args = winrt::make_self<implementation::GetWindowLayoutArgs>();
+        _GetWindowLayoutRequestedHandlers(nullptr, *args);
+        if (const auto op = args->WindowLayoutJsonAsync())
+        {
+            // This will fail if called on the UI thread, so the monarch should
+            // never set WindowLayoutJsonAsync.
+            auto str = op.get();
+            return str;

```

Comment:
You could make this function async by changing the interface definition to something like:
```idl
Windows.Foundation.IAsyncOperation<String> GetWindowLayout();
```

Reply:
Do async actions work over IPC? I kind of just assumed they wouldn't. I can try.

Reply:
(I haven't tried yet) The other side is that I don't want to have to make the monarch part async as well, if possible.

Reply:
It's my understanding of COM that it doesn't really care about process boundaries. As such async operations should work just fine.
But yeah please keep it as it is, if you'd like. _If you want to however_ you can do the synchronous `.get()` in the Monarch instead of the Peasant. This technically allows one to fetch the layout concurrently from all Peasants. But it's not of any importance for this PR.

Reply:
🤯 magic Windows object system. The whole concept of waiting on a coroutine in another process sounds crazy to me, but I guess that is the runtime's problem not mine.